detection_rules/beats.py

```diff
@@ -117,6 +117,13 @@ def _flatten_schema(schema: list, prefix="") -> list:
             # it's probably not perfect, but we can fix other bugs as we run into them later
             if len(schema) == 1 and nested_prefix.startswith(prefix + prefix):
                 nested_prefix = s["name"] + "."
+            if "fields" not in s:
+                # integrations sometimes have a group with no fields
+                continue
+            elif "field" in s:
+                # integrations sometimes have a group with a single field
+                flattened.extend(_flatten_schema(s["field"], prefix=nested_prefix))
```

Comment:
Design Q: Do we want to support integrations here or should this be moved to another method within `integrations.py`.

Reply:
We should definitely have a dedicated integrations file, but if we can _simply_ abstract the shared code, we should do that too. We can probably the shared code in beats and import to integrations.

For the sake of a POC though, I would duplicate the code to isolate differences.

But also, if these findings are accurate, looks like we need to update the beats generation code as well

Reply:
do you have an example to share for `field`

Reply:
<img width="1096" alt="Screenshot 2023-01-17 at 4 40 56 PM" src="https://user-images.githubusercontent.com/1636709/213018804-0cf2acb8-b13d-419c-b628-8fec3c252786.png">

- This is for instances where the `fields` key is not available.
 
<img width="736" alt="Screenshot 2023-01-17 at 4 43 25 PM" src="https://user-images.githubusercontent.com/1636709/213018800-07558bd4-ba05-4248-a462-bf999d760150.png">
- This is an example of when `field` is present.
- This is azure auditlogs v1.1.10